uber/models.py

```diff
@@ -1244,9 +1190,10 @@ def _badge_adjustments(self):
         self.badge_type = get_real_badge_type(self.badge_type)
 
         if not needs_badge_num(self):
-            if self.orig_value_of('badge_num') and c.SHIFT_CUSTOM_BADGES:
-                self.session.shift_badges(self.orig_value_of('badge_type'), self.orig_value_of('badge_num') + 1, down=True)
             self.badge_num = None
+
+        if self.orig_value_of('badge_type') != self.badge_type or self.orig_value_of('badge_num') != self.badge_num:
```

Comment:
This is what calls update_badge() for our change_badge page now. Plus, thanks to our setting the badge_num to `None` just above, this will also call update_badge for badges that are given INVALID status or changed by the attendee (which can be done in rare cases) or by other functions (like `unset_volunteering`, which is now called BEFORE this and does not itself call update_badge()).